### PR TITLE
Support GHC 8.8

### PR DESCRIPTION
**Pull request by fumieval**
Pardon an automatically-posted pull request; this is a part of [Operation Vanguard](https://github.com/haskell-vanguard/haskell-vanguard). Note that this does not guarantee that tests and benchmarks are buildable.

### CODE REVIEW

**Comment by hvr** [comment]
Tbh, an automatically filed PR whose only purpose is to relax upper bounds, without any human that has consciously reviewed the changelogs to verify that the relaxations are in fact sound/safe doesn't incite any confidence in me...

File: hackage-security/hackage-security.cabal [hunk 1]
```diff
@@ -119,7 +119,7 @@ library
                        cryptohash-sha256 >= 0.11    && < 0.12,
                        -- 0.4.2 introduces TarIndex, 0.4.4 introduces more
                        -- functionality, 0.5.0 changes type of serialise
-                       tar               >= 0.5     && < 0.6,
```
Comment (23Skidoo) [inline hackage-security/hackage-security.cabal]: GHC comes with `tar` now? Since when?

**Comment by 23Skidoo** [comment]
I tried to build `hackage-security` with a GHC 8.8.1 snapshot, and it failed because `dlist` doesn't work with `base-4.13`.

File: hackage-security/hackage-security.cabal (hunk 1 above)
Comment (hvr) [reply hackage-security/hackage-security.cabal]: it doesn't....

**Comment by achirkin** [comment]
I tried to build it today; the dependencies seems to be no longer a problem, but `MonadFail` is:
[`lookupM`](https://github.com/haskell/hackage-security/blob/474768743f407edef988d4153f081b2c662fe84f/hackage-security/src/Hackage/Security/TUF/FileMap.hs#L81) in `Hackage.Security.TUF.FileMap` requires it, which kind of contradicts to all `MonadError DeserializationError m` in other places.

**Pull request closed without merging**